baremetal/metal3remediation_manager_test.go

```diff
@@ -574,6 +574,12 @@ var _ = Describe("Metal3Remediation manager", func() {
 			},
 			RemediationTypeSet: false,
 		}),
+		Entry("Remediation strategy is nil should return empty string", testCaseGetRemediationType{
+			Metal3Remediation: &infrav1.Metal3Remediation{
+				Spec: infrav1.Metal3RemediationSpec{},
+			},
+			RemediationTypeSet: false,
+		}),
```

Comment:
This looks like a duplicate of the previous entry eventually (one omitting `Strategy` field at all and the other the `Type` inside the `Strategy` )?

Reply:
This test case is added to test the handling of the case where `Strategy` is nil. If there is no code added in the last commit, there will be a panic here.

Reply:
That is true, sorry I was wrong, this is a different test case actually